src/main/java/org/candlepin/service/model/Event.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+/**
+ * The Event interface represents a Candlepin event.
+ */
+public interface Event extends ServiceAdapterModel {
+
+    /**
+     * @return the event serialized in a format such as JSON.
+     */
+    String getBody();
+
+    /**
+     * @return the event's type
```

Comment:
Same here. When we're defining an interface, we should be very specific as to what is happening with regards to the input, output, and expected lifecycle or state management of the implementations.

Reply:
Updated to hopefully make the interface more clear